scripts/postinstall.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Generate preinstalled.json from folders in src/extensions/preinstalled-extensions
+ * ESM version to match repo "type": "module". Offline only.
+ */
+
+import { promises as fs } from 'node:fs';
+import fsSync from 'node:fs';
+import path from 'node:path';
+import crypto from 'node:crypto';
+import process from 'node:process';
+
+function generateSecureExtensionId(manifest) {
+  const payload = JSON.stringify({
+    name: manifest.name,
+    version: manifest.version,
+    description: manifest.description || '',
+    author: manifest.author || '',
+    homepage_url: manifest.homepage_url || ''
+  });
+  return crypto.createHash('sha256').update(payload).digest('hex').slice(0, 32);
+}
+
+async function main() {
+  try {
+    const root = process.cwd();
+    const preRoot = path.join(root, 'src', 'extensions', 'preinstalled-extensions');
+    if (!fsSync.existsSync(preRoot)) {
+      console.log('[postinstall] No preinstalled extensions directory found, skipping');
+      return;
+    }
+
+    const entries = [];
+    const dirents = await fs.readdir(preRoot, { withFileTypes: true });
+    for (const d of dirents) {
+      if (!d.isDirectory()) continue;
+      const dir = path.join(preRoot, d.name);
+      const manifestPath = path.join(dir, 'manifest.json');
```

Comment:
We should support multiple dynamic `manifest` file names like `manifest.chrome.json`, `manifest.firefox.json`, or `manifest.v3.json`.